src/rpc/mempool.cpp

```diff
@@ -605,7 +605,6 @@ static RPCHelpMan gettxspendingprevout()
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
         {
-            RPCTypeCheckArgument(request.params[0], UniValue::VARR);
```

Comment:
why is this redundant?

Reply:
oops, great catch. There is another one in coins.cpp. Pushing..

Reply:
Now you changed the error code from -3 to -1, effectively the opposite of what #26039 is doing

Reply:
true.. general exceptions are -1, while this is a type error.

We could throw a custom exception on the UniValue type check to catch all of them as type errors.

But.. I'm more inclined to close this PR and directly go ahead with moving the type checks responsibility to the request handler (which seems that is what you are doing in #26039).
Haven't checked if there is any restriction there but we do have each arg type on the help info object, so.. we could use that to verify every arg type.

Reply:
now you are changing the behavior in commit 8bdd192d4f897cc872f33cd6d15c36449dc345e5 and then reverting the behavior change in the next commit?